core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -242,6 +242,6 @@ private TableProperties() {
   public static final String MERGE_CARDINALITY_CHECK_ENABLED = "write.merge.cardinality-check.enabled";
   public static final boolean MERGE_CARDINALITY_CHECK_ENABLED_DEFAULT = true;
 
-  public static final String UPSERT_MODE_ENABLE = "write.upsert.enable";
-  public static final boolean UPSERT_MODE_ENABLE_DEFAULT = false;
+  public static final String UPSERT_ENABLED = "write.upsert.enabled";
```

Comment:
Given that this is a breaking change, should we allow both for at least one release?

Especially given that this could be set in a number of places such as:
- table properties
- at the job level
- anything at the job level could be set in the clusters config file

Reply:
We don't have any released version since the PR got merged: https://github.com/apache/iceberg/pull/2863.  I think it's OK to rename it to `enabled`. 

I think it's my mistake because I did not notice that all the iceberg switch are named as `xxx.enabled` before.  Thanks for the fix before we have a release, so that we don't have to maintain this inconsistent name for at least one release.

Reply:
Agree, I think we can just rename it as it hasn't been released.

Reply:
Ah that's great. If it hasn't been released then no worries. Retroactive +1.